src/libcore/cell.rs

```diff
@@ -188,6 +188,36 @@ use ptr;
 
 /// A mutable memory location.
 ///
+/// # Examples
+///
+/// Here you can see how using `Cell<T>` allows to use mutable field inside
+/// immutable struct (which is also called 'interior mutability').
+///
+/// ```
+/// use std::cell::Cell;
+///
+/// struct SomeStruct {
+///     regular_field: u8,
+///     special_field: Cell<u8>,
+/// }
+///
+/// fn main() {
+///     let my_struct = SomeStruct {
+///         regular_field: 0,
+///         special_field: Cell::new(1),
+///     };
+///
+///     let new_value = 100;
+///
+///     // ERROR, because my_struct is immutable
+///     // immutable.regular_field = new_value;
+///
+///     // WORKS, although `my_struct` is immutable, field `special_field` is mutable because it is Cell
+///     immutable.special_field.set(new_value);
```

Comment:
main isn't the issue. here's the error

```text
01:00:45] 	error[E0425]: cannot find value `immutable` in this scope

[01:00:45]   --> <anon>:22:1

[01:00:45]    |

[01:00:45] 22 | immutable.special_field.set(new_value);

[01:00:45]    | ^^^^^^^^^ not found in this scope

[01:00:45] 

[01:00:45] error[E0425]: cannot find value `immutable` in this scope

[01:00:45]   --> <anon>:23:12

[01:00:45]    |

[01:00:45] 23 | assert_eq!(immutable.special_field.get(), new_value);

[01:00:45]    |            ^^^^^^^^^ not found in this scope

[01:00:45] 
```

on this line, you use something called "immutable" but you never defined it; it's called `my_struct` above